src/test/java/io/zold/api/WalletsInTest.java

```diff
@@ -52,20 +62,45 @@ public void iteratesWallets() {
         MatcherAssert.assertThat(
             new WalletsIn(Paths.get("src/test/resources/walletsIn")),
             // @checkstyle MagicNumber (1 line)
-            Matchers.iterableWithSize(5)
+            new IsIterableWithSize<>(new IsEqual<>(5))
         );
     }
 
+    @Ignore("see @todo on WalletsIn.create")
     @Test
-    public void createIsNotYetImplemented() throws IOException {
-        this.thrown.expect(UnsupportedOperationException.class);
-        this.thrown.expectMessage(
-            Matchers.is(
-                "create() not yet supported"
-            )
+    public void createsWalletWithId() throws IOException {
+        final long id = 1L;
+        MatcherAssert.assertThat(
+            "Can't create wallet with id",
+            new WalletsIn(this.folder.newFolder().toPath()).create(id).id(),
+            new IsEqual<>(id)
+        );
+    }
+
+    @Test
+    public void createsWalletInWallets() throws IOException {
+        final Wallets wallets = new WalletsIn(this.folder.newFolder().toPath());
+        wallets.create(1L);
+        MatcherAssert.assertThat(
+            "Can't create wallet in wallets",
+            wallets,
+            new IsIterableWithSize<>(new IsEqual<>(1))
```

Comment:
@victornoel can we check the IDs of the returned wallets as well?

That said I know this might not be working yet, but let's add that to the puzzle to expand this test.

Reply:
@victornoel understood.